test_util/aws.py

```diff
@@ -83,21 +82,26 @@ def delete_key_pair(self, key_name):
         log.info('Deleting KeyPair: {}'.format(key_name))
         self.resource('ec2').KeyPair(key_name).delete()
 
-    def create_stack(self, name, template_url, parameters, deploy_timeout=60):
+    def create_stack(self, name, parameters, template_url=None, template_body=None, deploy_timeout=60):
         """Pulls template and checks user params versus temlate params.
         Does simple casting of strings or numbers
         Starts stack creation if validation is successful
         """
         log.info('Requesting AWS CloudFormation: {}'.format(name))
-        return self.resource('cloudformation').create_stack(
-            StackName=name,
-            TemplateURL=template_url,
-            DisableRollback=True,
-            TimeoutInMinutes=deploy_timeout,
-            Capabilities=['CAPABILITY_IAM'],
+        args = {
+            'StackName': name,
+            'DisableRollback': True,
+            'TimeoutInMinutes': deploy_timeout,
+            'Capabilities': ['CAPABILITY_IAM'],
             # this python API only accepts data in string format; cast as string here
             # so that we may pass parameters directly from yaml (which parses numbers as non-strings)
-            Parameters=param_dict_to_aws_format(parameters))
+            'Parameters': param_dict_to_aws_format(parameters)}
+        if template_body is not None:
+            assert template_url is None, 'tempate_body and template_url cannot be supplied simultaneously'
+            args['TemplateBody'] = template_body
+        else:
+            args['TemplateURL'] = template_url
```

Comment:
Should we check that one of these args passed? Looks like if I don't pass `template_url` or `template_body`, it'll fail with a type error for `template_url`, which could be confusing. "I didn't even pass template_url!"